### ISSUE TITLE
NULL value not assignable to Nullable<bool> parameter

### ISSUE THREAD

**Issue by SashaPshenychniy** [LightBDD/LightBDD#145]
#### Description
When TestCaseData passes NULL value for test argument of type Nullable<>, then exception is thrown 
```
System.InvalidOperationException : Provided argument  'null' is not assignable to parameter index 1 of method Void MyTestMethod(Boolean, System.Nullable`1[System.Boolean])
   at LightBDD.Core.Extensibility.ScenarioDescriptor.BuildParameters(MethodBase methodInfo, Object[] arguments)
```
Nullable structs should support setting NULL value to them.
The bug must be in [ScenarioDescriptor.cs](https://github.com/LightBDD/LightBDD/blob/master/src/LightBDD.Core/Extensibility/ScenarioDescriptor.cs) file (BuildParameters method).

#### Progress

- [x] Feature is implemented,
- [x] Ensured backward-compatibility,
- [ ] Ensured good debugging experience

**Comment by Suremaker** [comment]
@SashaPshenychniy can you please provide an example recreating the issue?
I will take a look at it in the evening.

**Comment by SashaPshenychniy** [comment]
Sorry, but for some reason I cannot reproduce this... I created few simple tests, even uncommented my old code and everything just works as expected now... Probably that was some stupid typo in my code.
But still, the following check look suspicious:
```
 if (argument == null && parameterType.IsValueType || argument != null && !parameterType.IsAssignableFrom(argument.GetType().GetTypeInfo()))
                    throw new InvalidOperationException($"Provided argument {argumentType} '{argument ?? "null"}' is not assignable to parameter index {i} of method {methodInfo}");
```

**Comment by Suremaker** [comment]
I was able to reproduce the issue and fixed it in LightBDD 2.4.3.
It was an edge case about passing `null` values to scenario methods accepting `Nullable<T>` parameters:

```c#
[Scenario]
[TestCase(1)]
[TestCase(null)]  // <--
public void Some_scenario(int? parameter)
{
    Runner.RunScenario(
        _ => Given_xxx(),
        _ => When_yyy(),
        _ => Then_zzz());
}
```

Interestingly, passing the `null` value to step method was working fine :)

Thanks @SashaPshenychniy for spotting it!
